src/lib/ui/components/Select/Select.svelte

```diff
@@ -3,82 +3,123 @@
+  let arrowElement: HTMLImageElement | null = null;
+  let errorElement: HTMLSpanElement | null = null;
+  let optionsElement: HTMLDivElement | null = null;
+
   let active = false;
+  let search = "";
 
-  const dispatch = createEventDispatcher<{ choose: string }>();
+  const dispatch = createEventDispatcher<{
+    choose: string[] | string;
+  }>();
   const form = useForm(context);
   const { data, errors, styles: ctxStyles, setField } = $form;
   $: ({ select: styles, icons, replace } = $ctxStyles);
 
-  function onChoose(value: string) {
+  function onChoose(value: string | string[]) {
     dispatch("choose", value);
   }
 
-  function handleToggle() {
+  function onToggle() {
     active = !active;
   }
 
+  function onHasValue() {
+    if (!multiple && $data[name]) {
+      const selection = options.find(({ value }) => value === $data[name]);
+      search = selection?.label ?? "";
+    }
+  }
+
+  function onFilter() {
+    if (!active) onToggle();
+  }
+
   function handleChoose(element: HTMLElement) {
-    const option =
-      element.role === "none" && element.tagName === "SPAN"
-        ? element.parentElement ?? element
-        : element;
+    const isOptions = element === optionsElement;
+    const option = isOptions
+      ? (element.firstElementChild as HTMLElement)
+      : element;
     const { value } = option.dataset;
     if (value) {
-      setField(name, value);
-      onChoose(value);
+      const selection = options.find(({ value: v }) => v === value);
+      const newValue = multiple
+        ? ([...($data[name] ?? []), value] as string[])
+        : value;
+      search = multiple ? "" : selection?.label ?? "";
+      setField(name, newValue);
+      onChoose(newValue);
     }
-    active = false;
+    active = multiple;
   }
 
   function handleSelect(event: MouseEvent) {
     const element = event.target as Target;
+    const isSelect = element === select;
+    const isLabel = element === labelElement;
+    const isContainer = element === container;
+    const isValueBox = element === valueBoxElement;
+    const isShowedOption =
+      element.tagName === tags.span &&
+      JSON.parse(element.dataset.optionShowed ?? "false");
+    const isSearchable = element === searchableElement;
+    const isArrow = element === arrowElement;
+    const isError = element === errorElement;
+    const isOptions = element === optionsElement;
+
     const isMenu =
-      element.role === "menu" ||
-      (element.role === "none" && element.parentElement?.role === "menu") ||
-      (element.role === "none" &&
-        (element.parentElement?.role === "presentation" ||
-          element.firstElementChild?.role === "presentation")) ||
-      (element.role === "presentation" &&
-        (element.parentElement?.role === "none" ||
-          element.firstElementChild?.role === "none"));
+      isSelect ||
+      isLabel ||
+      isContainer ||
+      isValueBox ||
+      isShowedOption ||
+      isSearchable ||
```

Comment:
🚫 **[eslint]** <[operator-linebreak](https://eslint.org/docs/latest/rules/operator-linebreak)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'||' should be placed at the beginning of the line.
```suggestion
      isShowedOption
      || isSearchable ||
```